types/yaireo__tagify/index.d.ts

```diff
@@ -0,0 +1,476 @@
+
+    /**
+     * Toggles class on the man tagify container (`scope`).
+     */
+    toggleClass(className: string, on: boolean): void;
+
+    /**
+     * Update `value` (array of objects) by traversing all valid tags.
+     */
+    updateValueByDOMTags(): void;
+
+    /**
+     * Converts a template string into a DOM node.
+     * @param template - Select a template from the `settings.templates` by name or supply a template function which returns a string
+     * @param data - Arguments passed to the template function
+     */
+    parseTemplate(template: 'wrapper', data: [HTMLInputElement | HTMLTextAreaElement, Tagify.TagifySettings]): HTMLElement;
+    parseTemplate(template: 'tag' | 'dropdownItem', data: [Tagify.TagData]): HTMLElement;
+    parseTemplate(template: 'dropdown', data: [Tagify.TagifySettings]): HTMLElement;
+    parseTemplate(template: 'dropdownItemNoMatch', data: []): HTMLElement;
+    parseTemplate(template: ((...args: any) => string), data: any[]): HTMLElement;
+
+    /**
+     * Toggles "readonly" mode on/off.
+     */
+    setReadonly(isReadonly: boolean): void;
+
+    /**
+     * Add event listener
+     */
+    on(event: 'add' | 'remove' | 'dblclick' | 'edit:beforeUpdate' | 'edit:updated', cb: (e: CustomEvent<{
+        tagify: Tagify, tag: HTMLElement, index?: number, data?: Tagify.TagData
+    }>) => void): this;
+    on(event: 'invalid', cb: (e: CustomEvent<{
+        tagify: Tagify, tag: HTMLElement, index?: number, data: Tagify.TagData, message: boolean
+    }>) => void): this;
+    on(event: 'input', cb: (e: CustomEvent<{
+        tagify: Tagify, value: string, inputElm: HTMLInputElement | HTMLTextAreaElement
+    }>) => void): this;
+    on(event: 'click', cb: (e: CustomEvent<{
+        tagify: Tagify, tag: HTMLElement, index: number, data: Tagify.TagData, originalEvent: MouseEvent
+    }>) => void): this;
+    on(event: 'keydown' | 'edit:keydown', cb: (e: CustomEvent<{ tagify: Tagify, originalEvent: KeyboardEvent }>) => void): this;
+    on(event: 'focus' | 'blur', cb: (e: CustomEvent<{
+        tagify: Tagify, relatedTarget: Element
+    }>) => void): this;
+    on(event: 'edit:start', cb: (e: CustomEvent<{
+        tagify: Tagify, tag: HTMLElement, index: number, data: Tagify.TagData, isValid: boolean
+    }>) => void): this;
+    on(event: 'edit:input', cb: (e: CustomEvent<{
+        tagify: Tagify, tag: HTMLElement, index: number, data: Tagify.TagData & { newValue: string }, originalEvent: Event
+    }>) => void): this;
+    on(event: 'dropdown:show' | 'dropdown:hide' | 'dropdown:updated', cb: (e: CustomEvent<{
+        tagify: Tagify, dropdown: HTMLElement
+    }>) => void): this;
+    on(event: 'dropdown:scroll', cb: (e: CustomEvent<{
+        tagify: Tagify, percentage: number
+    }>) => void): this;
+    on(event: 'dropdown:select', cb: (e: CustomEvent<{
+        tagify: Tagify, value: string;
+    }>) => void): this;
+}
+
+export = Tagify;
```

Comment:
Seems like this is exported using `export default` here: https://github.com/yairEO/tagify/blob/master/src/tagify.js#L1534

Reply:
I first did `export default Tagify;`, but the "CI / test" failed with the reason `NoDefaultExports`. That's why I changed it, after some research and local testing, to `export =`.

So, how to address default exports, such that DefinitelyTyped checks will be happy?

In this earlier version, I also exported the interfaces next to the default export:

```typescript
export interface TagifySettings { }
export interface TagData { }
export default class Tagify { }

// "CI / test" failed with NoDefaultExports
```

Reply:
Those interfaces need to be moved inside namespace Tagify

Reply:
I changed it back to `export default` with interfaces inside namespace:
```typescript
declare namespace Tagify {
    interface TagfifySettings { }
    interface TagData { }
}
declare class Tagfify { }
export default Tagify;
```
But now, the "CI / test" is failing again with `NeedsExportEquals` and `NoDefaultExport`. And since I'm not allowed to touch the `tslint.json` as suggested in the error log below, I don't know how to fix this issue.

I don't know how the "CI / test" works. But it obviously checks against the original library. If it uses the npm package, it will only find an UMD js file, i.e. a transpiled version of the source file and as such without any `default` keyword.

Any advice will be appreciated! (Thanks for your time btw.)

```
ERROR: 1:1    npm-naming  The declaration doesn't match the JavaScript module 'yaireo__tagify'. Reason:
The declaration should use 'export =' syntax because the JavaScript source uses 'module.exports =' syntax and 'module.exports' can be called or constructed.

To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
If you won't fix this error now or you think this error is wrong,
you can disable this check by adding the following options to your project's tslint.json file under "rules":

    "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
 See: https://github.com/Microsoft/dtslint/blob/master/docs/npm-naming.md

ERROR: 475:1  npm-naming  The declaration doesn't match the JavaScript module 'yaireo__tagify'. Reason:
The declaration specifies 'export default' but the JavaScript source does not mention 'default' anywhere.

The most common way to resolve this error is to use 'export =' syntax instead of 'export default'.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
If you won't fix this error now or you think this error is wrong,
you can disable this check by adding the following options to your project's tslint.json file under "rules":

    "npm-naming": [true,{"mode":"code","errors":[["NoDefaultExport",false]]}]
 See: https://github.com/Microsoft/dtslint/blob/master/docs/npm-naming.md
```

Reply:
Sorry i mispoke.. I didnt look at the unpkged file.. it does use `export=` https://unpkg.com/browse/@yaireo/tagify@3.22.1/dist/tagify.min.js 
So it does need to be `export=`. Sorry again for confusing.